app-api/src/main/java/org/openmrs/eip/app/CustomDatabaseOffsetBackingStore.java

```diff
@@ -0,0 +1,135 @@
+package org.openmrs.eip.app;
+
+import java.io.Serializable;
+import java.util.Date;
+import java.util.Map;
+
+import org.apache.commons.lang3.SerializationUtils;
+import org.apache.kafka.connect.runtime.WorkerConfig;
+import org.apache.kafka.connect.storage.Converter;
+import org.apache.kafka.connect.storage.MemoryOffsetBackingStore;
+import org.openmrs.eip.app.management.entity.DebeziumOffset;
+import org.openmrs.eip.app.management.repository.DebeziumOffsetRepository;
+import org.openmrs.eip.app.util.DebeziumOffsetUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.debezium.config.Instantiator;
+import io.debezium.embedded.EmbeddedEngine;
+
+@Component
+public class CustomDatabaseOffsetBackingStore extends MemoryOffsetBackingStore {
+	
+	private static final Logger log = LoggerFactory.getLogger(CustomDatabaseOffsetBackingStore.class);
+	
+	private static DebeziumOffsetRepository debeziumOffsetRepository;
```

Comment:
Why is this static?

Reply:
Resolved and replaced with the use of SyncContext (I wasn't aware of it's existence).